DOCS: Refactor (#878)
ansys/pymechanical: doc/source/faq.rst

    bring your own Python environment, which may contain other modules and tools. There is
    no dependency on opening the Mechanical user interface.

.. dropdown:: What is the relationship with Ansys Workbench?

    Ansys Workbench is a no-code environment to set up analysis systems that can be linked
    together. It is part of the Ansys family of software tools for process automation and design
    exploration. This family includes Ansys OptiSLang, which may be a more natural fit
    for integration with PyMechanical. The most popular app within the Workbench environment is
    Mechanical, and for many years, Workbench was the only environment you could run Mechanical from.

    Because Workbench is a no-code environment, a lot of the complexity around managing data
    transfer between Ansys apps and running parametric studies is hidden. PyMechanical and
    PyAnsys libraries more broadly give you much more control over your process automation and design
    exploration. However, eliminating Workbench means that you miss out on what it handled under
    the hood.

.. dropdown:: How do you restart a script?

>     If you have trouble terminating a simulation, you do not have to close Python, reopen it, and
>     clear all previous data such as the mesh using this code:

Reviewer: ```suggestion
    If you have trouble terminating a simulation, you do not have to close Python, reopen it, and
    clear all previous data, such as the mesh, using this code:
```

Reviewer: Is this within Workbench or a script that you would just run in the terminal while using pymechanical?

    .. code:: python

        import sys

        sys.modules[__name__].__dict__.clear()

    Simply exiting Python should clear the solution within Python. This is because
    stopping the original process means that nothing should be in present in
    a new process.

    The way that you clear all data from Mechanical in PyMechanical depends on if
    Mechanical is a remote session or embedded.

    - If Mechanical is a remote session, use either the
      `Mechanical.clear() <../api/ansys/mechanical/core/mechanical/Mechanical.html#Mechanical.clear>`_
      method or exit and restart Mechanical.
    - If Mechanical is embedded, use the
      `app.new() <../api/ansys/mechanical/core/embedding/app/App.html#App.new>`_
      method.

> .. dropdown:: How to check if license is active with PyMechanical ?

Reviewer: ```suggestion
.. dropdown:: How do you check if a license is active with PyMechanical?
```

>     Mechanical product related information can be printed with remote and embedding mode as given below

Reviewer: ```suggestion
    Information about Mechanical can be printed with remote and embedding mode:
```

    .. tab-set::

        .. tab-item:: Remote

            .. code-block:: python

                import ansys.mechanical.core as pymechanical

                mechanical = pymechanical.launch_mechanical()
                print(mechanical)

        .. tab-item:: Embedding

            .. code-block:: python

                import ansys.mechanical.core as pymechanical

                app = pymechanical.App()
                print(app)

>     The output from above shows what license is being used inside [] right to Ansys Mechanical . If the PyMechanical
>     is unable to grab any license then it will be left blank

Reviewer: ```suggestion
    The preceding code will print the license being used by Mechanical, the product version, and the software build date.
    The license is located within the brackets to the right of "Ansys Mechanical," as shown in the code block below. If the license does not exist, the brackets are empty.
```

    .. code-block:: shell

        Ansys Mechanical [Ansys Mechanical Enterprise]
        Product Version:242
        Software build date: 06/03/2024 09:35:09

> .. dropdown:: How to fix mechanical-env error in Linux ?

Reviewer: ```suggestion
.. dropdown:: How do you fix a ``mechanical-env`` error in Linux?
```

>     To use embedded instance in Linux, mechanical-env should be called before starting python shell or running python script.
>     Example usage is given below

Reviewer: ```suggestion
    To use an embedded instance in Linux, the ``mechanical-env`` command should be called before starting a Python shell or running a Python script:
```

.. _faq:

Frequently asked questions
==========================

This section provides answers to frequently asked questions.

.. dropdown:: How do you report issues?

>     :animate: fade-in-slide-down

Reviewer 2: Really cool feature!
